### DISCUSSION TITLE
Is there a way to detect if a `ContentDialog` is being displayed

### DISCUSSION THREAD

**Discussion by harvinders** [microsoft/microsoft-ui-xaml#8515] [Q&A]
I found that we can use the following to find out if there are any popups open. However, this API returns popup for `TeachingTip` etc too. Is there a way to find out if the Popup returned is a `ContentDialog`?

```cs
var openedpopups = VisualTreeHelper.GetOpenPopupsForXamlRoot(XamlRoot);
```

**Comment by castorix** [comment]
If I do a loop on openedpopups, I get _.Child_ = Microsoft.UI.Xaml.Controls.ContentDialog when a ContentDialog is displayed
(with Windows App SDK 1.1.0)

**Reply by harvinders** [reply]
I have following code, which does not work in 1.3.1

```cs
private async void AppWindow_Closing(AppWindow sender, AppWindowClosingEventArgs args)
{
  args.Cancel = true;

  var openedpopups = VisualTreeHelper.GetOpenPopupsForXamlRoot(XamlRoot);

  foreach (var openedpopup in openedpopups)
  {
    if (openedpopup.Child is ContentDialog) // Getting Child as Grid instead
    {
      return;
    }
  }
}
```

**Comment by Gaoyifei1011** [comment]
> 我发现我们可以使用以下方法来找出是否有任何弹出窗口打开。但是，此 API 也会返回等的弹出窗口。有没有办法找出返回的弹出窗口是否是？`TeachingTip``ContentDialog`
> 
> ```cs
> var openedpopups = VisualTreeHelper.GetOpenPopupsForXamlRoot(XamlRoot);
> ```

You can detect controls that the ContentDialog has but other pop-ups don't, such as Smokelayorbackground.

**Reply by harvinders** [reply]
```cs
private async void AppWindow_Closing(AppWindow sender, AppWindowClosingEventArgs args)
{
  args.Cancel = true;

  var openedpopups = VisualTreeHelper.GetOpenPopupsForXamlRoot(XamlRoot);

  foreach (var openedpopup in openedpopups)
  {
    if (openedpopup.Child is Rectangle) // This indicates the smoke layer, but it feels too hacky and would easily break in future versions.
    {
      return;
    }
  }
}
```
